From 17872ee9e9dace35d820086e7aa22158ff9ecce5 Mon Sep 17 00:00:00 2001
From: Wen Gong <quic_wgong@quicinc.com>
Date: Sat, 10 Sep 2022 11:43:50 -0400
Subject: [PATCH 103/112] wifi: mac80211: update all links rx nss by
 ieee80211_sta_set_rx_nss() in ieee80211_assoc_success()

Made changes to update all links rx nss by using
ieee80211_sta_set_rx_nss

Signed-off-by: Wen Gong <quic_wgong@quicinc.com>
Signed-off-by: Amutha Ravikumar <quic_aravikum@quicinc.com>
---
 net/mac80211/mlme.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/net/mac80211/mlme.c b/net/mac80211/mlme.c
index 517ad065118d..820db02342a1 100644
--- a/net/mac80211/mlme.c
+++ b/net/mac80211/mlme.c
@@ -5081,6 +5081,13 @@ static bool ieee80211_assoc_success(struct ieee80211_sub_if_data *sdata,
 	ieee80211_vif_set_links(sdata, valid_links);
 
 	rate_control_rate_init(sta);
+	for (link_id = 0; link_id < IEEE80211_MLD_MAX_NUM_LINKS; link_id++) {
+		if (!(sta->sta.valid_links & BIT(link_id)) ||
+		    sta->deflink.link_id == link_id)
+			continue;
+
+		ieee80211_sta_set_rx_nss(sta->link[link_id]);
+	}
 
 	if (ifmgd->flags & IEEE80211_STA_MFP_ENABLED) {
 		set_sta_flag(sta, WLAN_STA_MFP);
-- 
2.31.1

